### ISSUE TITLE
8.0.5 no longer lists ipykernel as an install dependency

### ISSUE THREAD

**Issue by stefanv** [jupyter-widgets/ipywidgets#3731]
This broke the scikit-image docs build.

See 785d1598b38f6fb43917489f530d14e28876fc24

**Comment by maartenbreddels** [comment]
Hi Stefan,

Yes, that's a feature. Since ipywidgets does not/should not make a decision on which kernel implementation to use.
Writing this out, I am unsure we should have done this in a fix release however.

Looking at a failure ( https://github.com/scikit-image/scikit-image/actions/runs/4482139559/jobs/7879781176?pr=6844 )
I am not sure if I'm happy with the situation either.
I think we should either have the comm package as a dependency or ipykernel, so ipywidgets works out of the box outside of a kernel.

cc @martinRenou

**Comment by larsoner** [comment]
I just hit the same bug in [MNE-Python tests](https://dev.azure.com/mne-tools/mne-python/_build/results?buildId=24337&view=logs&jobId=dded70eb-633c-5c42-e995-a7f8d1f99d91&j=dded70eb-633c-5c42-e995-a7f8d1f99d91&t=6a33d698-56da-505a-4bdd-64e49791e9d1). I plan to add `ipykernel` as a dependency in the same place as `ipywidgets` for now.

FWIW the error is cryptic:
```
_______________ ERROR collecting mne/viz/backends/_notebook.py ________________
mne\viz\backends\_notebook.py:53: in <module>
    _BASE_KWARGS = dict(layout=Layout(
C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\ipywidgets\widgets\widget_layout.py:86: in __init__
    super().__init__(**kwargs)
C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\ipywidgets\widgets\widget.py:504: in __init__
    self.open()
C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\ipywidgets\widgets\widget.py:535: in open
    self.comm = create_comm(**args)
C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\comm\__init__.py:27: in _create_comm
    raise NotImplementedError("Cannot ")
E   NotImplementedError: Cannot
```
So it would be nice to at least detect the problem and throw an informative error.

**Comment by maartenbreddels** [comment]
Yeah, that's not great. I think https://github.com/ipython/comm/pull/13 already solves most of the pain.

**Comment by jasongrout** [comment]
@maartenbreddels - I think you're right that we should have waited until 8.1 to make this change (especially looking at the number of issues linked here). What do you think of releasing an 8.0.6 with the dependency back in (so people can upgrade to the patch release with no breakage), and an 8.1.0 with the dependency removed, and noted in the release notes?

**Comment by GregIthaca** [comment]
For me this was showing up as the rather cryptic
```
jupyter_client.kernelspec.NoSuchKernel: No such kernel named python3
```
and it took a couple of hours to rebuild an environment where I could reproduce this. I might put something up on StackOverflow about this, since none of the existing questions seem to recognize this change as a possible cause.
